### ISSUE TITLE
Operation remove part does not return removed part

### ISSUE THREAD

**Issue by TLHeart** [simon-82/ModVarietyPack#85]
set specific operation to remove simple prosthesis, to get the prosthetic back... operation is performed, but no parts are returned.

**Comment by simon-82** [comment]
Yep, tried this now. Using the "remove part" operation does not spawn the part. 
Would be interesting if the parts get returned in vanilla. 

All I found was this one recipe in Core that deals with the removing of all parts in the game:

``` xml
<!-- removing body parts -->

  <RecipeDef ParentName="SurgeryFlesh">
        <defName>RemoveBodyPart</defName>
        <label>remove part</label>
        <description>Removes body part.</description>
        <workerClass>Recipe_RemoveBodyPart</workerClass>
        <jobString>Removing body part.</jobString>
        <workAmount>2000</workAmount>
        <ingredients>
            <li>
            <filter>
                <categories>
                <li>Medicine</li>
                </categories>
            </filter>
            <count>1</count>
            </li>
        </ingredients>
        <fixedIngredientFilter>
            <categories>
                <li>Medicine</li>
            </categories>
        </fixedIngredientFilter>
    </RecipeDef>
```

**Comment by TLHeart** [comment]
In vanilla with no mods, yes remove part does return an artificial part. just tested.

in EPOE when you install an artificial body part, and there is a natural body part, you will get the natural body part as part of the install... but the remove body part nets nothing.

Also missing from your version of EPOE is the ability to fix scars.

**Comment by simon-82** [comment]
Ok, interesting with vanilla. So you get the part back. Hm. Must have something to do with the worker class then. (workerClass Recipe_RemoveBodyPart) 
Maybe the vanilla prosthetics are hardcoded to that class. 
Would explain why EPOE does not give the modded part back without adding that to the class. 
Not sure about that, though...

Oh, fixing scars you say? I will have to look at EPOE again. Sounds nice. Do you happen to know where that function can be found in EPOE?

**Comment by TLHeart** [comment]
fix scars is in the def/recipesDefs/recipe_misc
 requires synthetic skin to also be manufactured.

synthetic skin is in the synthesizer recipes defs.

**Comment by simon-82** [comment]
Nice! mrofa was helping me on the slack to figure this out. 
Turns out there is an xml tag in the hediff givers that spawns the part back. EPOE was missing this.
I add it in to all parts, this should give back any part you remove

```
<spawnThingOnRemoved>AdvancedBionicEar</spawnThingOnRemoved>
```

I also tried out the scar removing. I had a colonist with an old gunshot and one with scratch scars.
Doing the operation with synthetic skin did nothing. 

I asked Ykara and it turns out removing scars does not work in Alpha 12. Last time it worked was in Alpha 10. So yea... I won't add that...
